volatility3/framework/plugins/linux/ebpf.py

```diff
@@ -0,0 +1,78 @@
+# This file is Copyright 2024 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+import binascii
+import logging
+from typing import List
+
+from volatility3.framework import renderers, interfaces, exceptions
+from volatility3.framework.objects import utility
+from volatility3.framework.renderers import format_hints
+from volatility3.framework.interfaces import plugins
+from volatility3.framework.configuration import requirements
+
+vollog = logging.getLogger(__name__)
+
+
+class EBPF(plugins.PluginInterface):
+    """Enumerate eBPF programs"""
+
+    _required_framework_version = (2, 0, 0)
+
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+        ]
+
+    def get_ebpf_programs(self, vmlinux) -> interfaces.objects.ObjectInterface:
+        """Enumerate eBPF programs walking its IDR.
+
+        Args:
+            vmlinux: The kernel symbols object
+
+        Yields:
+            eBPF program objects
+        """
+        if not vmlinux.has_symbol("prog_idr"):
+            raise exceptions.VolatilityException(
+                "Cannot find the eBPF prog idr. Unsupported kernel"
+            )
+
+        prog_idr_addr = vmlinux.get_symbol("prog_idr").address
+        prog_idr = vmlinux.object("idr", offset=prog_idr_addr)
+        for page_addr in prog_idr.get_page_addresses():
+            bpf_prog = vmlinux.object("bpf_prog", offset=page_addr, absolute=True)
+            yield bpf_prog
+
+    def _generator(self):
+        vmlinux = self.context.modules[self.config["kernel"]]
+        vmlinux_layer = vmlinux.context.layers[vmlinux.layer_name]
+        bpf_prog_types = vmlinux.get_enumeration("bpf_prog_type")
+        for prog in self.get_ebpf_programs(vmlinux):
+            prog_addr = prog.vol.offset
+            prog_type = bpf_prog_types.lookup(prog.type)
```

Comment:
I feel like you *should* just be able to ask `prog.type.lookup()` if prog of the enumeration type?  You could also do `prog.type.description` although it's technically more characters...

Reply:
Done. I found that I already added a bpf_prog extension object that returned the prog type when I wrote Sockstat. So, I ended up updating the ebpf extension objects. This will improve support with older kernels and avoid repeating code in future ebpf-related plugins